packages/blade/rollup.config.js

```diff
@@ -88,11 +88,23 @@ const getNativeConfig = ({ exportCategory }) => ({
   ],
 });
 
-// clean outputRootDirectory before building
-fs.rmSync(outputRootDirectory, { recursive: true, force: true });
+const getDeclarationsConfig = ({ exportCategory }) => ({
+  input: `${outputRootDirectory}/types/${exportCategory}/index.d.ts`,
+  output: [
+    {
+      file: `${outputRootDirectory}/${exportCategory}/index.d.ts`,
+      format: 'esm',
+    },
+  ],
+  plugins: [pluginDeclarations()],
+});
 
 const config = exportCategories
-  .map((exportCategory) => [getWebConfig({ exportCategory }), getNativeConfig({ exportCategory })])
+  .map((exportCategory) => [
+    getWebConfig({ exportCategory }),
+    getNativeConfig({ exportCategory }),
+    getDeclarationsConfig({ exportCategory }),
```

Comment:
we run `getDeclarationsConfig` after all source code is bundled.

Reply:
comment in the sense?

Reply:
I meant your original comment

> we run getDeclarationsConfig after all source code is bundled.

Can be added in code for reference

Reply:
Updated the comment. Does this makes sense? kept it simple. the previous one will be tied to RFC stuff